src/main/java/org/apache/commons/imaging/formats/webp/WebPImageParser.java

```diff
@@ -0,0 +1,351 @@
+                        formatDetails = "WebP/Lossless (Extended)";
+                    } else {
+                        throw new ImageReadException("Unknown webp chunk type: " + chunk);
+                    }
+                }
+            } else {
+                throw new ImageReadException("Unknown webp chunk type: " + chunk);
+            }
+
+            return new ImageInfo(
+                    formatDetails, 32, new ArrayList<>(), ImageFormats.WEBP,
+                    "webp", height, "image/webp", numberOfImages,
+                    -1, -1, -1, -1,
+                    width, false, hasAlpha, false,
+                    colorType, ImageInfo.CompressionAlgorithm.UNKNOWN
+            );
+        }
+    }
+
+    @Override
+    public Dimension getImageSize(ByteSource byteSource, WebPImagingParameters params) throws ImageReadException, IOException {
+        try (ChunksReader reader = new ChunksReader(byteSource)) {
+            WebPChunk chunk = reader.readChunk();
+            if (chunk instanceof WebPChunkVP8) {
+                WebPChunkVP8 vp8 = (WebPChunkVP8) chunk;
+                return new Dimension(vp8.getWidth(), vp8.getHeight());
+            } else if (chunk instanceof WebPChunkVP8L) {
+                WebPChunkVP8L vp8l = (WebPChunkVP8L) chunk;
+                return new Dimension(vp8l.getImageWidth(), vp8l.getImageHeight());
+            } else if (chunk instanceof WebPChunkVP8X) {
+                WebPChunkVP8X vp8x = (WebPChunkVP8X) chunk;
+                return new Dimension(vp8x.getCanvasWidth(), vp8x.getCanvasHeight());
+            } else {
+                throw new ImageReadException("Unknown webp chunk type: " + chunk);
+            }
+        }
+    }
+
+    @Override
+    public byte[] getICCProfileBytes(ByteSource byteSource, WebPImagingParameters params) throws ImageReadException, IOException {
+        try (ChunksReader reader = new ChunksReader(byteSource, WebPChunkXMP.TYPE_ICCP)) {
+            WebPChunk chunk = reader.readChunk();
+            return chunk == null ? null : chunk.getBytes();
+        }
+    }
+
+    @Override
+    public BufferedImage getBufferedImage(ByteSource byteSource, WebPImagingParameters params) throws ImageReadException, IOException {
+        throw new ImageReadException("Reading WebP files is currently not supported");
+    }
+
+    @Override
+    public boolean dumpImageFile(PrintWriter pw, ByteSource byteSource) throws ImageReadException, IOException {
+        pw.println("webp.dumpImageFile");
+        try (ChunksReader reader = new ChunksReader(byteSource)) {
+            int offset = reader.getOffset();
+            WebPChunk chunk = reader.readChunk();
+            if (chunk == null) {
+                throw new ImageReadException("No WebP chunks found");
+            }
+
+            do {
+                chunk.dump(pw, offset);
+
+                offset = reader.getOffset();
+                chunk = reader.readChunk();
+            } while (chunk != null);
```

Comment:
Adding a note to self: check for possible loops & add/multiply that could be made more defensive/safe to avoid CVE's (we had a few found by users & by the oss-fuzzer)

Reply:
Adding a `TODO` comment. Thought a little about it, and I think we are safe from DOS attacks here.